### DISCUSSION TITLE
2 Meters on 1 ESP32

### DISCUSSION THREAD

**Discussion by StarScream159** [tronikos/esphome-magnetometer-water-gas-meter#28] [Q&A] [answered]
Hello, is it possible to run 2 meters off a single ESP32? I would like to measure water and gas. I currently have water working on an ESP32 (calibrating right now) but I am getting values. Would it be possible to add a second meter on different pins on the ESP? How would I go about adding that to my existing config?

Thanks!

**Comment by tronikos** [comment, marked as answer]
I have https://github.com/tronikos/esphome-magnetometer-water-gas-meter/blob/main/esphome-two-meters.yaml but it's untested.
In your config replace `file: esphome-water-meter.yaml` with `file: esphome-two-meters.yaml` and set the `substitutions` to appropriate values.

**Reply by StarScream159** [reply]
This is great, thanks!

**Reply by StarScream159** [reply]
Looking at getting this configured now. I don't see hide_magnetic_field_strength_sensors / hide_half_rotations_total_sensor being passed to esphome-magnetometer.yaml from esphome-two-meters.yaml. If I set these substitutions will they still work? Sorry pretty new to esp stuff.

**Reply by tronikos** [reply]
No they won't. If you do need to show these sensors for calibration the easiest would be to just use the esphome-water-meter.yaml and esphome-gas-meter.yaml separately to find the values you need and then use esphome-two-meters.yaml. Alternatively, wait for me to find some time to add substitutions for these.
